src/components/ciConfig/CIConfigForm.tsx

```diff
@@ -73,8 +73,28 @@ export default function CIConfigForm({
                     (material) => material.id === ciConfig?.ciBuildConfig?.buildContextGitMaterialId,
                 )
               : sourceConfig.material[0]
-    const currentBuildContextGitMaterial = buildCtxGitMaterial || currentMaterial
-    const [selectedMaterial, setSelectedMaterial] = useState(currentMaterial)
+
+    const getParsedCurrentMaterial = (material?): SelectedGitMaterialType => {
+        const _currentMaterial: SelectedGitMaterialType = {
+            ...(material ? material : currentMaterial),
```

Comment:
Why are we spreading material and for label and value using currentMaterial. It can cause data discripencies in some case